src/script/lua_api/l_areastore.cpp

```diff
@@ -0,0 +1,420 @@
+/*
+Minetest
+Copyright (C) 2015 est31 <mtest31@outlook.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+
+#include "lua_api/l_areastore.h"
+#include "lua_api/l_internal.h"
+#include "common/c_converter.h"
+#include "cpp_api/s_security.h"
+#include "areastore.h"
+#include "filesys.h"
+#ifndef ANDROID
+#include "cmake_config.h"
+#endif
+#include <fstream>
+
+static inline void get_data_and_border_flags(lua_State *L, u8 start_i,
+		bool *borders, bool *data)
+{
+	if (!lua_isboolean(L, start_i))
+		return;
+	*borders = lua_toboolean(L, start_i);
+	if (!lua_isboolean(L, start_i + 1))
+		return;
+	*data = lua_toboolean(L, start_i + 1);
+}
+
+static inline void push_area(lua_State *L, const Area *a, bool borders, bool data)
+{
+	if (!(borders || data)) {
+		lua_pushboolean(L, true);
+	}
+	lua_newtable(L);
+	if (borders) {
+		lua_pushstring(L, "min");
+		push_v3s16(L, a->minedge);
+		lua_settable(L,-3);
+		lua_pushstring(L, "max");
+		push_v3s16(L, a->maxedge);
+		lua_settable(L,-3);
+	}
+	if (data) {
+		lua_pushstring(L, "data");
+		lua_pushlstring(L, a->data, a->datalen);
+		lua_settable(L,-3);
+	}
+}
+
+static inline void push_areas(lua_State *L, const std::vector<Area *> &areas,
+		bool borders, bool data)
+{
+	lua_newtable(L);
+	size_t cnt = areas.size();
+	for (size_t i = 0; i < cnt; i++) {
```

Comment:
I recognize your concern and can guess the reason why you separated areas.size() from the rest of the for loop here.
Since areas is const, the expression areas.size() would get evaluated only once had you written it in the "for (size_t i = 0; i < areas.size(); i++) {" manner.